src/main.py

```diff
@@ -262,6 +253,22 @@ def body(soup):
 def appendices(soup):
     return jats('appendices_json', base_url(jats('publisher_id')(soup)))(soup)
 
+def preprint_events(struct):
+    "returns a list of 'preprint' type events from article's pub-history or `None`"
+    if not struct or not isinstance(struct, list) or len(struct) == 0:
+        return
+    return [event for event in struct if event.get('type') == 'preprint'] or None
```

Comment:
I think possibly I named this `event_type`, i.e. `if event.get('event_type')`

Reply:
it's a good point, I'm not testing against anything coming from elife-tools. Let me double check what elife-tools is returning

Reply:
here is your work: https://github.com/elifesciences/issues/issues/6696#issuecomment-850580729-permalink

and it does indeed look like `event_type`, good catch

Reply:
test updated using actual output